src/BlockPatterns.php

```diff
@@ -48,6 +48,7 @@ public function __construct( Package $package ) {
 		$this->patterns_path = $package->get_path( 'patterns' );
 
 		add_action( 'init', array( $this, 'register_block_patterns' ) );
+		add_theme_support( 'woocommerce-blocks-patterns' );
```

Comment:
This should be invoked in a callback on the `after_setup_theme` action.

Reply:
> This should be invoked in a callback on the `after_setup_theme` action.

Even though that's the right thing to do, also supported by [docs](https://codex.wordpress.org/Plugin_API/Action_Reference):

> [after_setup_theme](https://codex.wordpress.org/Plugin_API/Action_Reference/after_setup_theme): add_theme_support() should be called here, since the init action hook is too late to add some features.

I think in this case we should call `add_theme_support` earlier. That's because:
- we call `add_theme_support` on `after_setup_theme` action
- we register patterns on `init` action

And that leaves no "right" moment/action for a theme to opt-out from the patterns.

In WordPress, the `add_theme_support('core-block-patterns')` is called even earlier - outside of class [here](https://github.com/WordPress/WordPress/blob/master/wp-includes/block-patterns.php#LL9C1-L9C44), so a user can opt-out by calling `remove_theme_support( 'core-block-patterns' );` from their `functions.php`.

Hence I think we should leave it that way or maybe even follow the WordPress approach. What do you think?

Reply:
Did you try hooking in on `after_setup_theme` on a early priority?

```php
add_action( 'after_setup_theme', function() {
	add_theme_support( 'woocommerce-blocks-patterns' );
}, 1 );
```
By registering our callback at an earlier priority, registration of a new callback on that action hook to remove will work with the default priority of 10.

Reply:
> so a user can opt-out by calling remove_theme_support( 'core-block-patterns' ); from their functions.php.

While allowing users to rely on `remove_theme_support` to disable WooCommerce block patterns is valid, IMO it is too restricted, as this function **should** be called in the theme's `functions.php` file, and child themes generally use it to override support from the parent theme.

With that said, I also recommend introducing our equivalent of the `should_load_remote_block_patterns` filter here for a more intuitive and straightforward opt-out (see comment below for more details).